### PR TITLE
refactor: Update Go version

### PR DESCRIPTION
**Pull request by alexlovelltroy**
This commit updates the Go version to the latest stable release and adds QEMU setup to the GitHub workflow. It also sets up the necessary environment variables required by GoReleaser. These changes ensure that the project is using the latest Go version and is ready for release.

Adds support for more architectures, package types, and better docker image.

### CODE REVIEW

File: .goreleaser.yaml [hunk 1]
```diff
@@ -7,12 +7,33 @@ before:
...
+         -X main.GitBranch={{.Branch}} \
+         -X main.GitTag={{.Tag}} \
+         -X main.GitState={{ .Env.GIT_STATE }} \
+         -X main.BuildHost={{ .Env.BUILD_HOST }} \
+         -X main.GoVersion={{ .Env.GO_VERSION }} \
+         -X main.BuildUser={{ .Env.BUILD_USER }} "
     goos:
       - linux
+      - darwin
+      - windows
```
Comment (davidallendj) [inline .goreleaser.yaml:28]: Was this intentional? This is the first time I'm Windows added to the `GOOS`.

File: cmd/version.go [hunk 2]
```diff
@@ -6,36 +6,69 @@ import (
...
+	fmt.Printf("Git State: %s\n", GitState)
+	fmt.Printf("Build Host: %s\n", BuildHost)
+	fmt.Printf("Go Version: %s\n", GoVersion)
+	fmt.Printf("Build User: %s\n", BuildUser)
+}
+
+func VersionInfo() string {
+	return fmt.Sprintf("Version: %s, Git Commit: %s, Build Time: %s, Git Branch: %s, Git Tag: %s, Git State: %s, Build Host: %s, Go Version: %s, Build User: %s",
+		Version, GitCommit, BuildTime, GitBranch, GitTag, GitState, BuildHost, GoVersion, BuildUser)
 }
```
Comment (davidallendj) [inline cmd/version.go:74]: Can this be moved to `internal/version/version.go` like in the other repos?

File: cmd/version.go [hunk 3]
```diff
@@ -6,36 +6,69 @@ import (
...
+	fmt.Printf("Build Time: %s\n", BuildTime)
+	fmt.Printf("Git Branch: %s\n", GitBranch)
+	fmt.Printf("Git Tag: %s\n", GitTag)
+	fmt.Printf("Git State: %s\n", GitState)
+	fmt.Printf("Build Host: %s\n", BuildHost)
+	fmt.Printf("Go Version: %s\n", GoVersion)
+	fmt.Printf("Build User: %s\n", BuildUser)
+}
+
+func VersionInfo() string {
```
Comment (davidallendj) [inline cmd/version.go:71]: Similar comment made in other repos: https://github.com/OpenCHAMI/bss/pull/48#discussion_r1822980267

**Review by davidallendj — CHANGES_REQUESTED** [review]
Couple of comments here, but I think moving the version implementation into `internal/version/version.go` is the most important one.

**Comment by alexlovelltroy** [comment]
Agreed about moving it. I'm actually working on exactly that right now. 

I left the VersionInfo string in the version.go file so that if we ever do use it, we have the same format in all software. 

People have asked for windows binaries before and it was pretty easy to add. We can remove if problematic.

**Comment by davidallendj** [comment]
> Agreed about moving it. I'm actually working on exactly that right now.
> 
> I left the VersionInfo string in the version.go file so that if we ever do use it, we have the same format in all software.
> 
> People have asked for windows binaries before and it was pretty easy to add. We can remove if problematic.

I don't think adding Windows binaries should be problematic, but it was the first time seeing it so I'm interested in seeing how that turns out.

As for the VersionInfo string, that seems to me like it would make sense in a common library but I don't think we currently have one. I think we also had a discussion before about trying to avoid having something like `hms-base` as well.

**Review by davidallendj — APPROVED** [review]
LGTM

**Pull request merged**